sql/core/src/test/scala/org/apache/spark/sql/connector/AlterTableTests.scala

```diff
@@ -101,6 +101,27 @@ trait AlterTableTests extends SharedSparkSession {
     }
   }
 
+  test("AlterTable: add column with position") {
```

Comment:
I think this should test failure cases as well, when the AFTER column does not exist for both nested and top-level additions.

Reply:
Can you also add a test case that hits the `span` bug above? That's where the AFTER column is not the first column in the struct. Adding `ADD COLUMN c AFTER b` should work.

Reply:
done